### ISSUE TITLE
vscode not detecting tests

### ISSUE THREAD

**Issue by 4rnaudB** [shimming-toolbox/shimming-toolbox#525]
### Description
I am currently using vscode version 1.90.0 and can no longer detect shimming-toolbox's tests.  It happened to me once before and I ran [this procedure](https://stackoverflow.com/questions/54387442/vs-code-not-finding-pytest-tests) and it fixed the problem, but not this time. Does anyone else using vscode have this problem? 
<img width="268" alt="image" src="https://github.com/shimming-toolbox/shimming-toolbox/assets/55560824/5c61e31d-d95e-45f3-b023-366e14114a7f">

When trying on SCT, I get the following error message:
```
2024-06-13 16:52:24.391 [error] Error discovering pytest tests:
 Error: spawn conda ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn conda',
  path: 'conda',
  spawnargs: [
    'run',
    '-n',
    'env_arnaud',
    '--no-capture-output',
    'python',
    '/Users/arnaud/.vscode/extensions/ms-python.python-2024.8.0/python_files/get_output_via_markers.py',
    '/Users/arnaud/.vscode/extensions/ms-python.python-2024.8.0/python_files/testing_tools/run_adapter.py',
    'discover',
    'pytest',
    '--',
    '--rootdir',
    '/Users/arnaud/Documents/spinalcordtoolbox/test',
    '-s',
    '--cache-clear',
    '.'
  ]
}
```

Also, @po09i doesn't have an issue with tests on pycharm.

**Comment by 4rnaudB** [comment]
Did you ever have a similar issue @mathieuboudreau?

**Comment by mathieuboudreau** [comment]
Hum, I don't think I ever used their addons to run the tests, sorry!

**Comment by mathieuboudreau** [comment]
What are the content/settings in the `.vscode` folder for your repo/repos?

**Comment by mathieuboudreau** [comment]
>When trying on SCT, I get the following error message:


What's the "Output" and "Test results" logs when you try executing your IVADOMED tests?

**Comment by mathieuboudreau** [comment]
I got them working on my end, so maybe we can compare logs/settings?

**Comment by 4rnaudB** [comment]
I can't execute the tests locally as they are not detected as tests by vs code. Vscode recognized there are tests in the repository because the testing tab is not available when there are no tests. 

For Ivadomed, I receive the same error message as sct:
2024-06-14 09:25:11.261 [error] Error discovering pytest tests:
 Error: spawn conda ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn conda',
  path: 'conda',
  spawnargs: [
    'run',
    '-n',
    'env_arnaud',
    '--no-capture-output',
    'python',
    '/Users/arnaud/.vscode/extensions/ms-python.python-2024.8.0/python_files/get_output_via_markers.py',
    '/Users/arnaud/.vscode/extensions/ms-python.python-2024.8.0/python_files/testing_tools/run_adapter.py',
    'discover',
    'pytest',
    '--',
    '--rootdir',
    '/Users/arnaud/Documents/ivadomed/test',
    '-s',
    '--cache-clear'
  ]
}

**Comment by 4rnaudB** [comment]
There was no vscode folder created in ivadomed, but in ST there is a launch.json and a settings.json file.

**Comment by joshuacwnewton** [comment]
I came across [this StackOverflow post](https://stackoverflow.com/questions/58799879/vscode-on-discover-tests-error-spawn-python-enoent) that suggests the error is due to an invalid path in `settings.json`.

Could you share the contents of `settings.json` (before making any changes, since I'm curious why the issue is occurring in the first place, hehe). Then, could you try some of the suggestions in that post?

**Comment by mathieuboudreau** [comment]
We solved the problem, the (main) issue was that there was a global root directory set for the tests in VScode (Code->Settings->Settings then search for "test" and find the python section),

![Screenshot 2024-06-14 at 11 46 27 AM](https://github.com/shimming-toolbox/shimming-toolbox/assets/1421029/79be09c6-5335-4eff-9dfb-6c7b4aa65a7d)

Removing this solved (most of) @4rnaudB 's issues.
![Screenshot 2024-06-14 at 11 46 33 AM](https://github.com/shimming-toolbox/shimming-toolbox/assets/1421029/1152e340-5f97-4f47-8818-1e78826e3cd3)

There were other minor venv related issues that we had to solve for specific projects, but that's not related specifically to this original issue.

**Comment by mathieuboudreau** [comment]
In addition, part of the reason that this worked previously *sometimes* for @4rnaudB is that some projects did have a "test" directory, and so by using that as the rootdir and then choosing "." as the folder for tests in the pytest extention, it would work.

**However**, ivadomed and SCT both use the directory *testing* and not *test*, thus repo/test/ did not exist for them, and that caused errors. Shimming-tooldbox used repo/test/.

https://github.com/ivadomed/ivadomed/tree/master/testing

https://github.com/spinalcordtoolbox/spinalcordtoolbox/tree/master/testing

**Comment by mathieuboudreau** [comment]
Lastly, we can see in the original post of this thread, it was trying to run the tests in the directory spinalcordtoolbox/test/, and that directory didn't exist (honestly though, VScode could have been more clear about the source of the error here).

![Screenshot 2024-06-14 at 12 09 34 PM](https://github.com/shimming-toolbox/shimming-toolbox/assets/1421029/401c7e45-8fd8-4239-b3a6-cb55b27cb600)

**Comment by mathieuboudreau** [comment]
It did say "ENOENT", and looking it up, that's what it means ("No such file or directory found"), but they could spell it out for us non-CS grads haha

**Comment by 4rnaudB** [comment]
For shimming-toolbox, the problem seemed to be related to the environment used. I duplicated shimming toolbox's environment maybe a year ago when St was still using Python 3.9. ST has now updated to Python 3.10.13 and my environment seems to be stuck at 3.9.x.  Even Conda seems to be confused since it doesn't which version of 3.9 to use (3.9.x). 
![image](https://github.com/shimming-toolbox/shimming-toolbox/assets/55560824/10d5f378-150d-4fd7-9787-6b8368c0fc01)

I cloned ST's environment with this command:
`conda create --clone /Users/arnaud/shimming-toolbox/python --name st_arnaud  `
